feat(employee registry) display stored date interval

jeremielodi opened the pull request:
This PR solves this problem #2228
closes #2228

jeremielodi committed:
feat(employee registry) display stored date interval

jniles commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -48,11 +49,13 @@ function bhDateInterval(moment, bhConstants) {
     startup();
   };
 
+  vm.onChange = function(){
```
This should be removed before merged into master.

jniles commented on client/src/modules/employees/registry/registry.js:
```diff
@@ -182,9 +182,11 @@ function EmployeeRegistryController(
-
     Employees.openSearchModal(filtersSnapshot)
       .then(function (changes) {
+        //This is very important if changes is undefined, a cache problem occurs
```
Nice comment. :+1:

jniles commented on client/src/modules/employees/registry/search.modal.js:
```diff
@@ -90,7 +91,7 @@ function EmployeeRegistryModalController(ModalInstance, bhConstants, moment, Ser
         // default to the original value if no display value is defined
         var displayValue = displayValues[key] || lastDisplayValues[key] || value;
-        changes.post({ key: key, value: value, displayValue: displayValue });
+        changes.post({ key: key, value: value, displayValue: value });
```
Is this change really necessary?  It looks like it undoes the work by @lomamech on displaying human-readable values instead of IDs.

jeremielodi replied:
I don't understand what is happening if I use  @lomamech  method here,  if I let  things like this
changes.post({ key: key, value: value, displayValue: displayValue });
the first store works but no the second.
When I try to change  doing things like this 
changes.post({ key: key, value: value, displayValue: value });
every thing works flawlessly

jniles replied:
Alright, I'll fork this PR and see if I can get it to work.  I don't want to destroy @lomamech's work.

jeremielodi replied:
I have already used @lomamech's  storing style in another issue, it worked without any problem. But I don't know why it poses problem here

jniles commented on client/src/modules/templates/bhDateInterval.tmpl.html:
```diff
@@ -1,4 +1,4 @@
-<ng-form name="DateIntervalForm" novalidate>
+<ng-form name="DateIntervalForm{{ $ctrl.dobDate }}" novalidate>
```
This looks like a typo.  Can you remove it?

jniles commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -79,27 +82,42 @@ function bhDateInterval(moment, bhConstants) {
   function startup() {
     var option;
-
+    /*
```
If these are no longer needed, just delete them.

jniles commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -79,27 +82,42 @@ function bhDateInterval(moment, bhConstants) {
   }
 
+ 
+  function ajustDateType(myvar){
```
Are you sure this function is necessary?  It looks like it will just do `new Date()` if `myvar` is defined.  what happens if you don't use it?

jeremielodi replied:
dates are not displayed misfortunatelly

jniles reviewed (dismissed):
There's a quite a few clean up things to take care of before this gets into `master`.  Let me know what you think.

jniles committed:
fix(employees): preserve search displayValues

This commit cleans up the bhDateInterval controller, and stored the
hiring dates in the displayValues once they change.  This allows them to
use the displayValue mechanism by @lomamech.
